app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt

```diff
@@ -498,6 +506,15 @@ class MainActivity : AppCompatActivity(), ColorPickerDialogListener {
         }
     }
 
+    override fun onStart() {
+        val settingsManager = PreferenceManager.getDefaultSharedPreferences(this)
+        super.onStart()
+        ioSafe {
+            if (githubApi.getLatestLoginData() != null && settingsManager.getBoolean(getString(R.string.automatic_cloud_backups), true)){
```

Comment:
This can still cause a race condition :pensive: 
I have thought a while and the best way I can think of resolving this is

1. Create a private variable `val accountsLoginLock = Mutex()`
2. [Here](https://github.com/recloudstream/cloudstream/blob/b6868217a03b0e0f8755fc933b65302fc9e60095/app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt#L610) do `githubLoginLock.withLock { ` after the ioSafe
3. Do the same [here](https://github.com/recloudstream/cloudstream/blob/b6868217a03b0e0f8755fc933b65302fc9e60095/app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt#L512)

The idea behind this is that when you do:

`githubLoginLock.withLock { /* first code here*/ }`

on one thread and then call another piece of code like this while the first runs:

`githubLoginLock.withLock { /* second code here*/ }`

then the second code will only be launched when the first code is completed.
You want to do this as otherwise the restore code can be launched before the app is fully logged in to github, which will fail the `githubApi.getLatestLoginData() != null` check and therefore not restore anything only because it was too fast.